File: app/src/main/java/kagura/project/com/a8/database/DAOBase.java
package kagura.project.com.a8.database;


import android.content.Context;
import android.database.sqlite.SQLiteDatabase;

class DAOBase {

    private final static int VERSION = 2;
    private final static String NOM = "results.db";
    private DatabaseHandler mHandler = null;
    SQLiteDatabase mDb = null;

    DAOBase(Context context){
        this.mHandler = new DatabaseHandler(context, NOM, null, VERSION);
    }

    public SQLiteDatabase open(){
        mDb = mHandler.getWritableDatabase();
        return mDb;
    }
}

File: app/src/main/java/kagura/project/com/a8/database/DAOBase.kt
package kagura.project.com.a8.database

import android.content.Context
import android.database.sqlite.SQLiteDatabase

open class DAOBase(context: Context?) {
    private var mHandler: DatabaseHandler? = null

    @JvmField
    var mDb: SQLiteDatabase? = null
    fun open(): SQLiteDatabase? {
        mDb = mHandler!!.writableDatabase
        return mDb
    }

    companion object {
        private const val VERSION = 2
        private const val NOM = "results.db"
    }

    init {
        mHandler = DatabaseHandler(context, NOM, null, VERSION)
    }
}